mdn/src/test/java/org/apache/james/mdn/MDNFactoryTest.java

```diff
@@ -330,88 +330,6 @@ public void generateMDNReportShouldFormatNoModifier() {
     }
 
     @Test
-    public void generateMDNReportShouldFormatNoModifierNullType() {
```

Comment:
you are missing tests for null/absent cases, showing that it throws?
please don't forget that you should throw when generating the report, not when populating the data structure, else you won't be able to read bad data when you implement a parser

Reply:
Not if the lenient parser corrects the value ;-)

Reply:
Such test are BTW implemented in https://github.com/linagora/james-project/pull/951/commits/937b6950c22c8abb00b75409c0d7ed6be24ae970